### DISCUSSION TITLE
How to integrate with ChatGPT (no code experience)

### DISCUSSION THREAD

**Discussion by jdizzle1236** [Glench/ExtPay#134] [Q&A]
i have no idea how to hire someone, cant find any reasonably priced people on fiverr and upwork doesnt let me post a job. i have a simple extention, checking for text and plays a sound when its found. i know how to make extension pay work using chatgpt to help me code since i dont know how to code my self. the issue is the extension works automatically and the extension pay doesnt block the main function.

**Comment by burkybang** [comment]


**Reply by jdizzle1236** [reply]
thanks for the thought, i just figured it out. ill write in the comments my process in case someone else finds it helpful

**Comment by jdizzle1236** [comment]
Heres my chatgpt process. chatgpt has limits to send messaged, for it to reply. work arounds are installing an extention (open source code) that can feed it txt files instead, so it will quickly break a long message into multiple parts to save time feeding chatgpt data. 50k max i think, you can easily ask what the limits are to chatgpt. it gets less accurate over several thousand characters. best to ask for small bits and pieces. can sometimes rename icon text and other text that is custom labels (label means text that isnt exactly code language). there are more in depth tutorials on youtube. anyways i tell it to combine background.js with the same file in the example extension, same with the manifest file. i had help building my extension feeding the robot open source code but i dunno if i actually used it i had to learn what was the best way to do what i wanted to do. combining those two files makes it working after you put in your extension id in the two needed files (shown in the extensionpay tutorial). so pretty easy to get it to work but if your extension functions while it hasnt been paid, this next part makes it so a part wont work unless user paid. 

ask chatgpt "what file would i use this in
"extpay.getUser().then(user => {
  if (user.paid) {
     mainFunction()
  }
})"
as im trying to make sure that the sound isnt played unless the user has paid "

sometimes its best to ask it to name files, as sometimes it only gives example code, so just ask for updated code for a specific file and it will, just will have to do that for all files sometimes. sometimes it misses the files needed. youll figure it out if it aint too hard for the robot to do, can ask for alternative ways of doing things or ways to fix errors and sometimes that works.
